src/control/server/ctl_system.go

```diff
@@ -75,10 +73,14 @@ func (svc *ControlService) getMSMember() (*system.Member, error) {
 	return msMember, nil
 }
 
+type temporary interface {
+	Temporary() bool
+}
+
 func isUnreachableError(err error) bool {
-	switch ne := errors.Cause(err).(type) {
-	case *net.OpError:
-		return !ne.Temporary()
+	te, ok := err.(temporary)
```

Comment:
I think you still need to unwrap the err with errors.Cause() here... If the root error was wrapped then this type assertion won't work.